tools/amd_build/pyHIPIFY/hipify-python.py

```diff
@@ -0,0 +1,967 @@
+#!/usr/bin/python
+""" The Python Hipify script.
+##
+# Copyright (c) 2015-2016 Advanced Micro Devices, Inc. All rights reserved.
+#               2017-2018 Advanced Micro Devices, Inc. and
+#                         Facebook Inc. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+
+import argparse
+import constants
+import re
+import shutil
+import sys
+import os
+import yaml
+
+from functools import reduce
+from enum import Enum
+from cuda_to_hip_mappings import CUDA_TO_HIP_MAPPINGS
+
+
+def openf(filename, mode):
+    if sys.version_info[0] == 3:
+        return open(filename, mode, errors='ignore')
```

Comment:
Previous comment was:

> This should not be necessary. All of the files in our repository should be UTF-8 encoded. Is this not true?

This is still not addressed.

Reply:
Sorry, yes. There is one file @jithunnair-amd located that is not UTF-8 encoded. @jithunnair-amd could you elaborate here?

Reply:
```
rocm-user@480e3f4175c2:~/pytorch__ROCM__master$ git diff --cached torch/csrc/jit/passes/shape_analysis.cpp
diff --git a/torch/csrc/jit/passes/shape_analysis.cpp b/torch/csrc/jit/passes/shape_analysis.cpp
index be18bf1..4c24c5c 100644
--- a/torch/csrc/jit/passes/shape_analysis.cpp
+++ b/torch/csrc/jit/passes/shape_analysis.cpp
@@ -325,7 +325,7 @@ void PropagateShapeOnNode(Node * node, bool insert_expands) {
    case prim::ReplaceIfUndef: {
      // If types[0] has a type, then it is not defined, and the type will
      // get set to types[0] because that will be the value propagated.
-      // If its type is<C2><A0>not<C2><A0>defined, then unification is an undefined type.
+      // If its type is not defined, then unification is an undefined type.
      SHAPE_ASSERT(types.size() == 1);
      node->output()->setType(types.at(0)->shared_from_this());
      handled = true;
```

Reply:
My master doesn't show anything here? https://github.com/pytorch/pytorch/blob/ea1869244ff84b256dd1f0b4de20fb604ec92bec/torch/csrc/jit/passes/shape_analysis.cpp#L328